cirq/sim/density_matrix_simulator_test.py

```diff
@@ -327,7 +327,7 @@ def test_run_measurement_not_terminal_no_repetitions(dtype):
                     result.measurements, {'0': np.empty([0, 1]), '1': np.empty([0, 1])}
                 )
                 assert result.repetitions == 0
-        assert mock_sim.call_count == 0
+        assert mock_sim.call_count == 4
```

Comment:
Why did this increase? For `repetitions=0`, I wouldn't expect `mock_sim` to be called.

Similarly for the other changes in this file.